Snake/SnakeTest/Food_Test.cs

```diff
@@ -7,37 +8,25 @@ namespace SnakeTest
     public class Food_Test
     {
         [TestMethod]
-        public void generateFoodTestOutsideBoundaryXAxis()
+        public void generateFoodTypeIsDeterministic()
         {
-            BusinessLayer.Food food = new BusinessLayer.Food();
+            Food food;
+            FoodGenerator foodGenerator = new FoodGenerator();
 
-            bool result = food.generateFood(-1, 40);
-
-            Assert.IsFalse(result, "Accepted Invalid X input");
-
-        }
-
-        [TestMethod]
-        public void generateFoodTestOutsideBoundaryYAxis()
-        {
-            BusinessLayer.Food food = new BusinessLayer.Food();
-
-            bool result = food.generateFood(26, -3);
-
-
-            Assert.IsFalse(result, "Accepted Invalid Y input");
+            food = foodGenerator.generateFood(1,1);
 
+            Assert.IsTrue(food is BasicFood || food is AdvancedFood, "Food type not determined");
         }
 
         [TestMethod]
-        public void generateFoodTestInsideBoundaryYAxis()
+        [ExpectedException(typeof(SystemException),

```

Comment:
👍 for testing for exceptions.

Not sure of the setup you guys are using - if you can install nuget packages you could consider using [FluentAssertions](http://www.fluentassertions.com/) which has a nicer way of [asserting exceptions](https://github.com/dennisdoomen/FluentAssertions/wiki#exceptions)

Probably out of scope though :)